pkg/quay/quay.go

```diff
@@ -0,0 +1,166 @@
+package quay
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	publicImagesOnly   = true
+	registryEndpoint   = "https://quay.io"
+	repositoryEndpoint = "https://quay.io/api/v1/repository"
+
+	tagLengthLimit = 20
+)
+
+type Quay struct {
+	namespace    string
+	lastModified time.Duration
+
+	httpClient *http.Client
+}
+
+type Config struct {
+	Namespace    string
+	LastModified time.Duration
+}
+
+type Repository struct {
+	Name         string `json:"name"`
+	LastModified int    `json:"last_modified"`
+}
+
+type RepositoriesJSON struct {
+	Repositories []Repository `json:"repositories"`
+}
+
```

Comment:
I'd prefer to move `Repository` & `RepositoriesJSON` to `types.go`.

```suggestion

type Repository struct {
	Name         string `json:"name"`
	LastModified int    `json:"last_modified"`
}

type RepositoriesJSON struct {
	Repositories []Repository `json:"repositories"`
}

type Config struct {
	Namespace    string
	LastModified time.Duration
}

type Quay struct {
	namespace    string
	lastModified time.Duration

	httpClient *http.Client
}

```